src/Microsoft.AspNet.Hosting.WindowsService/WindowsServiceWebApplication.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Microsoft.AspNet.Hosting.Internal;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using System.ServiceProcess;
+
+namespace Microsoft.AspNet.Hosting.WindowsService
+{
+    public class WindowsServiceWebApplication : ServiceBase
+    {
+        private const string HostingJsonFile = "hosting.json";
+        private const string EnvironmentVariablesPrefix = "ASPNET_";
+        private const string ConfigFileKey = "config";
+        private IApplication _application;
+        private bool _stopRequestedByWindows;
+
+        public static void Run()
+        {
+            Run(new WindowsServiceWebApplication());
+        }
+
+        public static void Run(WindowsServiceWebApplication app)
+        {
+            ServiceBase.Run(app);
+        }
+
+        protected sealed override void OnStart(string[] args)
+        {
+            OnStarting(args);
+
+            // Allow the location of the json file to be specified via a --config command line arg
+            var tempConfigBuilder = new ConfigurationBuilder().AddCommandLine(args);
+            var tempConfig = tempConfigBuilder.Build();
+            var configFilePath = tempConfig[ConfigFileKey] ?? HostingJsonFile;
+            var config = CreateConfiguration(configFilePath, args);
+            var hostBuilder = new WebHostBuilder(config, captureStartupErrors: true);
```

Comment:
I think `captureStartupErrors` should be configurable, and **off** by default. I'd rather my Windows service crash on start and have the exception logged via the usual failed service mechanism than report that it started and have the error only be available by hitting the endpoint and getting HTML. I believe if we just throw the exception here, `ServiceBase` will do the right thing and the log the exception as part of the failed to start message in the Windows Event Log.